### DISCUSSION TITLE
Some wicked "scenarios" for easy server set up

### DISCUSSION THREAD

**Discussion by noway** [piqueserver/piqueserver#637] [Feature Requests]
This one actually would give this project the most bang for the buck.

While it had been quite a good decision to threw out some unneeded maps/scripts out of PySnip to make the distribution leaner, the amount of time to get the server enjoyable became too much. You need to go online/to BnS/to aloha forums and search for the scripts and maps to fill your config file. And you really need to have some solid AoS experience to weed out good maps and good scripts.

What we can do:
A directory with "scenarios" which would hold easy to install "bundles" of config files: scripts and maps already in them. Some of the possible "scenarios":

1. Several rounds of good hallway/flatmap-style shoot-and-dig game
2. Arena gamemode with all the best arena maps
3. Fully set-up and ready-to-play  nuketown madness
4. Babel and the best maps with it
5. Etc, etc.

System administrator user journey would be like this:
1. Go to the "scenario directory" web site
2. Look at the scenarios, their picture/description
3. Copy the scenario name and execute something along the lines of `piqueserver piqueserver/babel`
4. Invite friends, connect and enjoy

Server creation would become as easy as it never has been before which is a plus. And it instantly makes potential openspades integration much more exciting if that's on the roadmap.

**Comment by NotAFile** [comment]
...because the only thing a gameserver is missing is a package manager!

I really like the idea of installing maps this way. This makes it way easier for creators to distribute and also update their maps.

Other projects have used github for simple package management before and it seems to work great!

**Comment by 1AmYF** [comment]
During development of the "Custom Game" software (a web-based server launcher) i compiled a list of gamemodes and corresponding maps, trying to represent both what was considered popular and balanced (and also visually appealing) at the time. Beside my own selection i also took the input from a bunch of experienced players and mappers and lateron also watched on maps that have been uploaded to the software numerous times. 
Maybe that can serve as a starting point for bundles/a package manager. That software has a page that prints out all modes & maps here http://bit.ly/2lAPI8M (it's currently not hosted on buildandshoot)

**Comment by noway** [comment]
Yes, I was inspired by exactly that service of yours! That looks like a solid range so far, so yeah, could be a nice starting point.

**Comment by samuelallan72** [comment]
Further discussion of this should take place in [piqueserver/piqueserver-extras](https://github.com/piqueserver/piqueserver-extras). :)

**Comment by noway** [comment]
awesome
